core/server/master/src/main/java/alluxio/master/block/RegisterStreamObserver.java

```diff
@@ -94,6 +94,9 @@ public RegisterWorkerPResponse call() throws Exception {
                 + "Probably %s was exceeded!",
             PropertyKey.MASTER_WORKER_REGISTER_STREAM_RESPONSE_TIMEOUT.toString());
 
+        // Update the TS before and after processing the request, so that when a message
+        // takes long to process, the stream does not time out.
+        mContext.updateTs();
         mBlockMaster.workerRegisterStream(mContext, chunk, isHead);
         mContext.updateTs();
```

Comment:
not familiar with this part of the code, but is the updateTs() call still necessary after the register?

Reply:
From the PR description,
> This change updates the timestamp more often.

So I believe the answer is yes; we want to update the timestamp before & afterwards in case the register takes an exceedingly long time.

Reply:
Yes. The `workerRegisterStream` can take much time in processing. If we only update TS at the end of a message processing, the interval equals `Tmessage_process + Tmessage_in_flight` where `Tmessage_in_flight` equals network delay and worker-side preparation time. If we update TS before and after, the interval becomes more like `max(Tmessage_process, Tmessage_in_flight)`.